surf/apps/communities/admin.py

```diff
@@ -4,31 +4,57 @@
 
 from django.contrib import admin
 from django import forms
-from django.db.models import Q
 from django.core.files.images import get_image_dimensions
 
 from surf.apps.communities import models
+from surf.apps.communities.models import PublishStatus
+
+
+def trash_nodes(modeladmin, request, queryset):
+    for obj in queryset:
+        obj.delete()
+
+
+trash_nodes.short_description = "Trash selected %(verbose_name_plural)s"
+
+
+def restore_nodes(modeladmin, request, queryset):
+    for obj in queryset:
+        obj.restore()
+
+
+restore_nodes.short_description = "Restore selected %(verbose_name_plural)s"
+
+
+class TrashListFilter(admin.SimpleListFilter):
+
+    title = 'is trash'
+    parameter_name = 'trash'
+
+    def lookups(self, request, model_admin):
+        return (
+            ('1', 'Yes'),
+            ('0', 'No'),
+        )
+
+    def queryset(self, request, queryset):
+        value = self.value() or '0'
+        try:
+            is_trash = bool(int(value))
+        except ValueError:
+            is_trash = False
+        return queryset.filter(deleted_at__isnull=not is_trash)
 
 
 class CommunityForm(forms.ModelForm):
     """
     Implementation of Community Form class.
     """
+    publish_status = forms.TypedChoiceField(choices=PublishStatus.choices(), coerce=int)
```

Comment:
Not sure, but can we choose coerce=string or something to prevent storing small integers? It saves a few bytes, but readability is very low for numbers

Reply:
The django_enumfield is a wrapper around an integerfield, so changing this to coerce=str doesn't change how the data is stored.